Buildings/Controls/OBC/CDL/Logical/VariablePulse.mo

```diff
@@ -0,0 +1,329 @@
+within Buildings.Controls.OBC.CDL.Logical;
+block VariablePulse
+  "Generate boolean pulse with the width specified by input"
+
+  parameter Real samplePeriod(unit="s")
+    "Sample period of the width input";
+  parameter Real period(unit="s")
+    "Time for one pulse period";
+  parameter Real chaWidThr=0.05
```

Comment:
Maybe this should be "Minimum required change in `u` to re-trigger an update of the output interval`

Reply:
updated.